File: Book/Fortran/Fig_4.3_parReg1.f90
Program parReg1 
   use omp_lib
  implicit none

   integer :: ID, size_of_team, NThrds
   call omp_set_num_threads(4)                                              
   !$omp parallel private(ID,NThrds)
      ID = omp_get_thread_num()
      NThrds = omp_get_num_threads()
      if (ID == 0) size_of_team = NThrds
   !$omp end parallel
  print *, "We just did the join on a team of size ", size_of_team
end Program parReg1

File: Book/C/Fig_4.3_parReg1.c
#include <stdio.h>
#include <omp.h>

int main()
{
  omp_set_num_threads(4);
  int size_of_team;
  #pragma omp parallel
  {
    int ID = omp_get_thread_num();  
    int NThrds = omp_get_num_threads();
    if (ID == 0) size_of_team = NThrds;
  } // end of parallel region
  printf("We just did the join on a team of size \%d", size_of_team);
}
